src/analysis/retail/warlock/demonology/modules/features/SummonDemonicTyrant.tsx

```diff
@@ -125,23 +198,264 @@ class SummonDemonicTyrant extends Analyzer {
       </table>
     );
 
-    const avgDemonsEmpowered = Number(this._summsWithDemonicPower[0]['Total']);
-    const avgWildImpsEmpowered = Number(this._summsWithDemonicPower[0]['Wild Imp']);
+    return petTable;
+  }
+
+  getImpsPerformance(imps: number): QualitativePerformance {
+    if ((this._hasReignOfTyranny && imps === 15) || imps === 10) {
+      return QualitativePerformance.Perfect;
+    }
+
+    if (imps >= 7) {
+      return QualitativePerformance.Good;
+    }
+
+    if (imps >= 3) {
+      return QualitativePerformance.Ok;
+    }
+
+    return QualitativePerformance.Fail;
+  }
+
+  getPerformanceExplanation(
+    actualPerformance: QualitativePerformance,
+    dogs: boolean,
+    gfg: boolean,
+    imps: number,
+  ): JSX.Element {
+    const impsExtended = `${imps}/${this._hasReignOfTyranny ? '15' : '10'} imps`;
+    switch (actualPerformance) {
+      case QualitativePerformance.Perfect:
+        return <>Perfect usage, you extended the most demons possible</>;
+      case QualitativePerformance.Good:
+        return <>Good usage, you extended {impsExtended}</>;
+      case QualitativePerformance.Ok:
+        return <>Bad usage, you extended {impsExtended}</>;
+      default: // QualitativePerformance.Fail
+        return dogs && gfg ? (
+          <>Terrible usage, you only extended ${impsExtended}</>
+        ) : dogs ? (
+          <>
+            Terrible usage, you missed <SpellLink spell={TALENTS.GRIMOIRE_FELGUARD_TALENT} />
+          </>
+        ) : (
+          <>
+            Terrible usage, you missed <SpellLink spell={TALENTS.CALL_DREADSTALKERS_TALENT} />
+          </>
+        );
+    }
+  }
+
+  calcTyrantUsage(tyrantCastNum: number): SpellUse {
+    const castEvent = this._tyrantCasts[tyrantCastNum];
+    const checklistItems: ChecklistUsageInfo[] = [];
+
+    // ------------------------------ Dogs -------------------------------------
+    const goodDogs = this._summsWithDemonicPower[tyrantCastNum]['Dreadstalker'] > 1;
+    const dogsPerformance = goodDogs ? QualitativePerformance.Perfect : QualitativePerformance.Fail;
+
+    const dogsSummary = (
```

Comment:
This method is very large, I'd suggest a splitting it up into a new method for each performance item to break this up a bit. Each can return the checklist item. As it is it's very large, and comments are not a good way of breaking it up.